nixos/modules/services/misc/duckdns.nix

```
@@ -0,0 +1,67 @@
+{ config, pkgs, lib, ... }:
+let
+cfg = config.services.duckdns;
+in
+{
+  options = {
+    services.duckdns = {
+      enable = lib.mkEnableOption (lib.mdDoc "DuckDNS Dynamic DNS Client");
+
+      tokenFile = lib.mkOption {
+        default = null;
+        type = lib.types.str;
+        description = lib.mdDoc ''
+          The path to a file containing the token
+          used to authenticate with DuckDNS.
+          '';
+      };
+
+      domains = lib.mkOption {
+        default = null;
+        type = lib.types.nullOr (lib.types.listOf lib.types.str);
+        example = [ "examplehost" ];
+        description = lib.mdDoc ''
+          The record(s) to update in DuckDNS
+          (without the .duckdns.org prefix)
+          '';
+      };
+
+      domainsFile = lib.mkOption {
+        default = null;
+        type = lib.types.nullOr lib.types.str;
+        description = lib.mdDoc ''
+          The path to a file containing a
+          newline-separated list of DuckDNS
+          domain(s) to be updated
+          '';
+      };
+
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
```

I think this can use an assertion to make sure either `domains` or `domainsFile` are set
```suggestion
  config = lib.mkIf cfg.enable {
    assertions = [
      {
        assertion = (cfg.domains != null || cfg.domainsFile != null);
        message = "services.duckdns.domains or services.duckdns.domainsFile has to be defined";
      }
    ];
```